src/lib/email-server.ts

```diff
@@ -0,0 +1,94 @@
+import { Server } from 'http';
+import express from 'express';
+import nodemailer from 'nodemailer';
+import { portFinder } from './port-finder';
+
+class EmailServer {
+	private server: Server | null = null;
+	private port: number | null = null;
+
+	async start(): Promise< number > {
+		if ( this.server ) {
+			return this.port!;
+		}
+
+		const app = express();
+		app.use( express.json() );
+
+		// @TODO: service should be configurable via Studio settings.
```

Comment:
The default would be Ethereal catch all for all sites. 

Should sites be able to use separate email services? For example, a user might want site `x`  to use gmail and not care about other sites? 

Or should settings affect all sites?